### PR TITLE
Update god-mode recipe to darth10/god-mode repo

### PR DESCRIPTION
**Pull request by darth10**
### Brief summary of what the package does

God-like command entering minor mode.

### Direct link to the package repository

Old: https://github.com/chrisdone/god-mode
New: https://github.com/emacsorphanage/god-mode 

### Your association with the package

Contributor. I'm also the author of the [`god` module in doom-emacs](https://github.com/hlissner/doom-emacs/tree/develop/modules/editor/god) that uses `god-mode`.

### Relevant communications with the upstream package maintainer

~I've tried reaching out to @chrisdone through email and keybase, but haven't received a response from him yet.~
As #6520 mentions, I believe @chrisdone is happy to have someone else to maintain this package.

@chrisdone has transferred the repo to emacsorphanage.

### Checklist

Please confirm with `x`:

- [x] The package is released under a [GPL-Compatible Free Software License](https://www.gnu.org/licenses/license-list.en.html#GPLCompatibleLicenses).
- [x] I've read [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [ ] I've used the latest version of [package-lint](https://github.com/purcell/package-lint) to check for packaging issues, and addressed its feedback
- [x] My elisp byte-compiles cleanly
- [ ] `M-x checkdoc` is happy with my docstrings
- [x] I've built and installed the package using the instructions in [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [ ] I have confirmed some of these without doing them

This PR updates the repo for this recipe to emacsorphanage/god-mode.

All of the code changes are cosmetic in nature to satisfy `M-x checkdoc` and `M-x package-lint-current-buffer`.
In other words, none of the code changes modify existing functionality.
Also, lexical binding has been enabled in all files.
Other changes include README updates, switching to Github actions for CI and moving relevant files to the `.github` directory.

#6520 is a similar attempt to update this recipe, but has functional enhancements in it as well.
I am more than happy to incorporate those enhancements in my fork once this recipe change is merged.

EDIT: All of the above changes will have to done in emacsorphanage/god-mode as it has been transferred.

### CODE REVIEW

**Comment by tarsius** [comment]
Please consider transferring @chrisdone to the [Emacs orphanage](https://github.com/emacsorphanage). I can then figure out who to make a maintainer and/or proxy maintain.

If you want to do that, then I have to temporarily make you a member of that organization so that you can transfer the repository in.

**Comment by darth10** [comment]
Hi @tarsius! 
Can we get this merged as it's been a while since we've both reached out to @chrisdone?

**Comment by tarsius** [comment]
@chrisdone one reason for why it would be better if you could transfer the repository is that then counting of issues would not start from `#1` again and it would never be ambiguous which issue a number refers to.

Above I mentioned the `emacsorphanage` but an easier approach would be to transfer my user account instead. You can initiate that without any prior action by me (i.e. now). I will still have to accept the transfer, but you are done after this single step.

Thanks!

**Comment by chrisdone** [comment]
Quoth Github:

"You don’t have the permission to create public repositories on emacsorphanage"

**Comment by tarsius** [comment]
I have now invited you to become a temporary member of the emacsorphanage. Once you have accepted you will be able to transfer the repository directly.

**Comment by darth10** [comment]
I've edited the original description now that god-mode has been transferred to emacsorphanage.
I also understand this PR may not be needed as GitHub will automatically perform redirects.
